sale_general_discount/models/sale.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+from lxml import etree
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    general_discount = fields.Float(
+        digits=dp.get_precision('Discount'),
+        string='Discount (%)',
+    )
+
+    @api.multi
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        for order in self:
+            super(SaleOrder, self).onchange_partner_id()
```

Comment:
You can use `super()` directly, it's Python 3!

OTOH, I don't know if you realized you're calling `super()` inside a loop! :speak_no_evil:  Put the call before entering the loop or we could get a bottleneck here.

Reply:
ups!!